### PR TITLE
Make pressing Enter confirm project creation/import in the project manager

### PR DESCRIPTION
**Pull request by Calinou**
This makes the project manager more friendly to keyboard usage.

You can now create projects more easily without touching the mouse by opening the project manager, pressing <kbd>Ctrl + N</kbd> (shortcut for the **Create** button), entering a project name and pressing <kbd>Enter</kbd>.

### CODE REVIEW

File: doc/classes/LineEdit.xml [hunk 1]
```diff
@@ -365,7 +365,7 @@
 		<signal name="text_submitted">
 			<param index="0" name="new_text" type="String" />
 			<description>
-				Emitted when the user presses [constant KEY_ENTER] on the [LineEdit].
```
Comment (Mickeon) [inline doc/classes/LineEdit.xml:368]: This change is unrelated *technically*

If we want to improve upon this description  , this is still too broad and should probably mention `ui_text_submit`, but again, if I want to be nitpicky, it's unrelated.

Comment (Calinou) [reply doc/classes/LineEdit.xml:368]: I've split this change to its own PR: https://github.com/godotengine/godot/pull/97455

File: editor/project_manager/project_dialog.cpp [hunk 2]
```diff
@@ -1024,8 +1024,14 @@ ProjectDialog::ProjectDialog() {
 	add_child(fdialog_install);
 
 	project_name->connect(SceneStringName(text_changed), callable_mp(this, &ProjectDialog::_project_name_changed).unbind(1));
+	project_name->connect(SceneStringName(text_submitted), callable_mp(this, &ProjectDialog::ok_pressed).unbind(1));
```
Comment (Maran23) [inline editor/project_manager/project_dialog.cpp:1027]: I would recommend to rather use the `register_text_enter` method for those three `LineEdit`s, e.g. here just call `register_text_enter(project_name);`.

This will effectively do the same, but also the disablement of the ok button is checked, so that `ok_pressed` is only called when it is enabled.

Comment (Calinou) [reply editor/project_manager/project_dialog.cpp]: That makes more sense indeed. I tested it and it works, thanks for the tip :slightly_smiling_face:

**Review by Maran23 — APPROVED** [review]
Code looks good and makes sense. A nice improvement for usability. 
Also matches pretty much all other dialogs, as I also identified/unified in https://github.com/godotengine/godot/pull/96400.

The `text_submitted` `SceneStringName` could be used/replaced everywhere else in a future PR.

**Comment by Calinou** [comment]
> The `text_submitted` `SceneStringName` could be used/replaced everywhere else in a future PR.

I removed it from this PR as it's no longer needed here.

**Review by radiantgurl — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
